### PR TITLE
746 improve package extraction error handling

### LINKED ISSUE #746: Error messages are unhelpful when installer fails to extract files
Consider this scenario where I don't have enough disk space to install a package:

This is the verbose log:
image

In many cases, a retry is completely useless, and in the case of missing space, it would be easy to add a prompt to free up space. If we are running non-interactively in this scenario, retrying is a complete waste of time.

In any case, we should log the exception instead of silently ignoring it.

### PR DESCRIPTION
**Pull request by alnlarsen**
Add retry dialog when out of space.
Also improve general error messages during package extraction

Closes #746

### CODE REVIEW

File: Package/Installer.cs [hunk 1]
```diff
@@ -405,7 +405,6 @@ enum AbortOrRetryResponse
     }
 
     [Obfuscation(Exclude = true)]
-    [Display("Package files are in use")]
     class AbortOrRetryRequest
```
Comment (rmadsen-ks) [inline Package/Installer.cs:408]: Now you removed this, but does that not affect the original use of the code?

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
Really nice improvement. I had one comment though.

File: Package/Installer.cs (hunk 1 above)
Comment (alnlarsen) [reply Package/Installer.cs:408]: You're right, I forgot to add a generic title :)

File: Package/Installer.cs [hunk 2]
```diff
@@ -274,7 +274,7 @@ private void WaitForPackageFilesFree(string tapDir, List<string> packagePaths)
 
                 while (isPackageFilesInUse(tapDir, packagePaths, exclude))
                 {
-                    var req = new AbortOrRetryRequest(inUseString) {Response = AbortOrRetryResponse.Abort};
+                    var req = new AbortOrRetryRequest("Package files are in use", inUseString) {Response = AbortOrRetryResponse.Abort};
```
Comment (rmadsen-ks) [inline Package/Installer.cs:277]:
```suggestion
                    var req = new AbortOrRetryRequest("Package files are in use.", inUseString) {Response = AbortOrRetryResponse.Abort};
```

I guess we should use either title case or proper punctuation here.

File: Package/PluginInstaller.cs [hunk 3]
```diff
@@ -455,7 +455,16 @@ internal static List<string> UnpackPackage(string packagePath, string destinatio
                             {
                                 throw;
                             }
-                            catch
+                            catch (IOException ex) when (ex.Message.Contains("There is not enough space on the disk"))
+                            {
+                                log.Error(ex.Message);
+                                var req = new AbortOrRetryRequest("Not enough disk space", $"File '{part.FullName}' requires {Utils.BytesToReadable(part.Length)} of free space. " +
```
Comment (rmadsen-ks) [inline Package/PluginInstaller.cs:461]:
```suggestion
                                var req = new AbortOrRetryRequest("Not Enough Disk Space", $"File '{part.FullName}' requires {Utils.BytesToReadable(part.Length)} of free space. " +
```

Here we can use title case.

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
What do you think about these minor changes?

**Comment by alnlarsen** [comment]
Let's use title casing in both dialogs then

**Review by rmadsen-ks — DISMISSED** [review]
Looks great!

**Comment by AsgerIversen** [comment]
Just thinking about how this will behave when the REST-API session manager is installing stuff for a new session... 

If on one is subscribed to the UserInput SignalR thing, I think the response will now just be "User aborted installation." when the real reason is "There is not enough space on the disk"

@Denrask maybe you can comment?

**Comment by alnlarsen** [comment]
That's a good point. How does REST handle the dialog when package files are in use? @Denrask @AsgerIversen 

In any case, I think the exception should mention that the installation was aborted due to missing disk space.

**Comment by Denrask** [comment]
I've been sylting this for a while. I'm not too concerned with this. I'll just handle it with default behavior.

**Review by Denrask — APPROVED** [review]

**Pull request merged**